### DISCUSSION TITLE
[V3] Automatically running properties validation rules

### DISCUSSION THREAD

**Discussion by DanielSpravtsev** [livewire/livewire#6157] [7. Bugs (archived)] [answered]
Hey,

Livewire docs says:

> Livewire will automatically run the properties validation rules before each update. This frees you from needing to run $this->validate() manually

Is it working only with wire:model.live directives? 

Because, when I click save, I getting Post create error without field validation...

```
Example from https://livewire.laravel.com/docs/validation#rule-attributes

use Livewire\Attributes\Rule;
use Livewire\Component;
use App\Models\Post;
 
class CreatePost extends Component
{
    #[Rule('required|min:3')] 
    public $title = '';
 
    #[Rule('required|min:3')] 
    public $content = '';
 
    public function save()
    {
        Post::create([
            'title' => $this->title,
            'content' => $this->content,
        ]);
 
        return redirect()->to('/posts');
    }
 
    // ...
}
```

**Comment by joshhanley** [comment]
@DanielSpravtsev thanks for reporting! Any chance you can submit a failing test PR so we can investigate? Thanks!

**Reply by edalzell** [reply]
What's weird is that there IS a test, and it passes: https://github.com/livewire/livewire/blob/11c75143c19d4f4559d0cf67e27f458cbd0c154e/src/Features/SupportValidation/UnitTest.php#L17

But still fails for  me on beta.8

**Reply by edalzell** [reply]
right so it ONLY fails when you use the front end. OK so need dusk test

**Comment by ZackBz** [comment]
I am also experiencing this when trying to validate a simple form. I even copy and pasted the [example from the docs ](https://livewire.laravel.com/docs/forms#adding-validation) and my form would submit without running any validation.  If i added `$this->validate()` to the save function  that is in the docs, then it would run the validation.

**Reply by reppair** [reply]
Same.

**Reply by joshhanley** [reply, marked as answer]
@ZackBz and @reppair the docs are wrong, you do still need to add `$this->validate();` to your save method. Any PRs to update it would be appreciated! 🙂

**Reply by reppair** [reply]
@joshhanley Isn't it supposed to work as documented instead?

I don't really care if I have to call the method or not, Livewire is doing so much for us already that this is not an issue at all.

All I am wondering here, is if this is the way to address this (update the docs), you know, considering @calebporzio's presentation at Laracon and reading the same in the docs, maybe, the behavior should change instead? Again, I don't know anything about such plans what so ever, just what I am thinking.

**Reply by reppair** [reply]
For others that may be getting confused around this, like I was, in `v2`'s docs we find this useful bit of information 😅 

> Please note that mutating a property directly inside a Livewire component class doesn't trigger any of the updating/updated hooks.

So, not knowing the above at the time, I was thinking, let's see if my prop even gets updated (so any alleged 'automatic' validation should run on the new value) so I started dumping in `updated()` / `updating()` which is pointless since the hooks won't run on internal updates. And as seen in the example, it's clearly meant to be used with `wire:model`.

<img width="739" alt="Screenshot 2023-08-21 at 10 11 55" src="https://github.com/livewire/livewire/assets/10784954/baa665f3-8ce4-476d-a21d-78c20af3c23e">

<img width="732" alt="Screenshot 2023-08-21 at 10 09 19" src="https://github.com/livewire/livewire/assets/10784954/c6895c2e-9846-4e11-9d3f-7982aae71cf3">
